src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiAddProjectPropertiesStep.cs

```diff
@@ -145,16 +148,33 @@ protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgrad
 
             var project = context.CurrentProject.Required();
             var file = project.GetFile();
+            var components = await project.GetComponentsAsync(token).ConfigureAwait(false);
+            bool propertiesValueUpdated = true;
 
-            // var properties =  project.GetProjectPropertyElements();
-            if (string.IsNullOrEmpty(file.GetPropertyValue("UseMaui")))
+            // check project properties updated based on project types
+            if (components.HasFlag(ProjectComponents.Maui))
```

Comment:
can `components` be `null` here?

Reply:
Oh no no, I updated that check, if the MAUI extension is running, the component value will be one of the 3 types, it won't be null